scripts/delete-federation.sh

```diff
@@ -24,23 +24,38 @@ set -o pipefail
 source "$(dirname "${BASH_SOURCE}")/util.sh"
 
 function delete-script-deployment() {
-  # Remove cluster registry CRD
-  ${KCD} -f vendor/k8s.io/cluster-registry/cluster-registry-crd.yaml
+  if [[ ! "${NAMESPACED}" || "${DELETE_CLUSTER_RESOURCE}" ]]; then
+      # Remove cluster registry CRD
+      ${KCD} -f vendor/k8s.io/cluster-registry/cluster-registry-crd.yaml
+  fi
 
   # Remove public namespace
   if [[ ! "${NAMESPACED}" ]]; then
     ${KCD} namespace ${PUBLIC_NS}
   fi
 
   # Disable federation of all types
-  for ftc in $(kubectl get federatedtypeconfig -n "${NS}" -o=jsonpath={.items..metadata.name}); do
-    ./bin/kubefed2 disable "${ftc}" --delete-from-api --federation-namespace="${NS}"
-  done
+  if [[ ! "${NAMESPACED}" ]]; then
+    for ftc in $(kubectl get federatedtypeconfig -n "${NS}" -o=jsonpath={.items..metadata.name}); do
+      ./bin/kubefed2 disable "${ftc}" --delete-from-api --federation-namespace="${NS}"
+    done
+  else
+    if [[ "${DELETE_CLUSTER_RESOURCE}" ]]; then
+      ${KCD} crd $(kubectl get crd | grep -E 'types.federation.k8s.io' | awk '{print $1}')
+    fi
+  fi
 
   # Remove federation CRDs, namespace, RBAC and deployment resources.
   if [[ ! "${USE_LATEST}" ]]; then
     if [[ "${NAMESPACED}" ]]; then
-      ${KCD} -n "${NS}" -f hack/install-namespaced.yaml
+      if [[ "${DELETE_CLUSTER_RESOURCE}" ]]; then
+        ${KCD} -n "${NS}" -f hack/install-namespaced.yaml
+      else
+        ${KCD} -n "${NS}" role federation-role
```

Comment:
Is it necessary to delete these resources explicitly vs deleting the containing namespace?